sklearn/metrics/tests/test_pairwise_distances_reduction.py

```diff
@@ -0,0 +1,466 @@
+import numpy as np
+import pytest
+from collections import defaultdict
+from numpy.testing import assert_array_equal, assert_allclose
+from scipy.sparse import csr_matrix
+
+from sklearn.metrics._dist_metrics import (
+    DenseDenseDatasetsPair,
+)
+
+from sklearn.metrics._pairwise_distances_reduction import (
+    PairwiseDistancesReduction,
+    PairwiseDistancesArgKmin,
+    FastEuclideanPairwiseDistancesArgKmin,
+    _sqeuclidean_row_norms,
+)
+
+from sklearn.utils import _in_unstable_openblas_configuration
+from sklearn.utils.fixes import sp_version, parse_version
+from sklearn.utils._testing import fails_if_unstable_openblas
+
+
+def _get_dummy_metric_params_list(metric: str, n_features: int):
```

Comment:
This function generates arrays even when it does not need it. For example, when `metric="euclidean"`, `V`, `weights`, and `VI` are create, but `[{}]` is returned.

Reply:
I'll address it in one of the dedicated sub-PRs.

Reply:
Adressed in https://github.com/scikit-learn/scikit-learn/pull/22064 via https://github.com/scikit-learn/scikit-learn/commit/3cdd3a5ad099326b1bd2c8601d7360272c10f7e4.